### DISCUSSION TITLE
Project .gitconfig

### DISCUSSION THREAD

**Discussion by alimd** [AliMD/alwatr#56] [Ideas]
به نظرتون چه کانفیگ هایی در گیت این پروژه بزاریم؟
و آیا واقعا همه اینا تو پروژه باشه کار میکنه؟ تست کنید

**Comment by alimd** [comment]
This is mine.

```gitconfig
[alias]
  c = commit -m
  ca = commit -am
  cl = clone
  cl1 = clone --depth=1
  pl = pull --prune --progress --autostash
  plr = pull --prune --progress --autostash --rebase
  p = push
  pp = push --prune
  pa = push --all
  pt = push --tags
  b = branch
  ba = branch -avv
  bd = push origin --delete
  s = status -s
  m = merge
  r = rebase
  rs = reset
  co = checkout
  dev = checkout develop
  f = flow
  chp = cherry-pick
  d = diff --color --color-words --abbrev
  l = log --color --graph --pretty=format:'%Cred%h%Creset -%C(yellow)%d%Creset %s %Cgreen(%cr) %C(bold blue)<%an>%Creset' --abbrev-commit --
  # Show the diff between the latest commit and the current state
  df = !"git diff-index --quiet HEAD -- || clear; git --no-pager diff --patch-with-stat"
  ru = rebase-update # depot_tools specific
  t = tag -ln3

  hist = log --pretty=format:\"%h %ad | %s%d [%an]\" --graph --date=short
  type = cat-file -t
  dump = cat-file -p

[help]
  autocorrect = 1

[push]
  default = current

[pull]
  rebase = false

# so much color
[color]
  ui = always

[color "diff"]
  meta = yellow bold
  commit = green bold
  frag = magenta bold
  old = red bold
  new = green bold
  whitespace = red reverse

[color "diff-highlight"]
  oldNormal = red bold
  oldHighlight = "red bold 52"
  newNormal = "green bold"
  newHighlight = "green bold 22"

[color "branch"]
  current = yellow reverse
  local = yellow
  remote = green

[color "status"]
  added = yellow
  changed = green
  untracked = cyan

[core]
  autocrlf = input

[init]
  defaultBranch = main

[commit]
  gpgsign = true
```

**Reply by alimd** [reply]
This is important and necessary
`autocrlf = input`

**Reply by alimd** [reply]
@asmanp2012

**Reply by alimd** [reply]
@MM25Zamanian

**Comment by alimd** [comment]
چرا انقدر یک ساده رو کفت میدین ؟!!!

**Reply by MM25Zamanian** [reply]
تستش کردم نمیشه اینطوری باید حتما به گیت بگیم از این کانفیگ خاص استفاده کنه وگرنه از کانفیگ خود سیستم استفاده میکنه

**Reply by alimd** [reply]
چجوری باید بگی؟!

**Reply by alimd** [reply]
میشه فقط جاش تو .git هست پس نمیشه کامیت بشه
به نظرم میشه تو دستورات yarn بزاریم 

https://stackoverflow.com/questions/8801729/is-it-possible-to-have-different-git-configuration-for-different-projects

**Reply by alimd** [reply]
و یا حداقل در ریدمی توضیح بدیم

**Reply by MM25Zamanian** [reply]
> چجوری باید بگی؟!

`git config -f .gitconfig`

**Reply by MM25Zamanian** [reply]
> میشه فقط جاش تو .git هست پس نمیشه کامیت بشه به نظرم میشه تو دستورات yarn بزاریم
> 
> https://stackoverflow.com/questions/8801729/is-it-possible-to-have-different-git-configuration-for-different-projects

موافقین بعد از نصب باشه که اگر پکیجی نصب شد این تنظیم رو انجام بده ؟؟

**Reply by alimd** [reply]
ربطی به نصب نداره
فرقی نداره

**Reply by alimd** [reply]
درضمن -f در اون فایل ذخیره می‌کنه نه اینکه از اون فایل بخونه

**Reply by MM25Zamanian** [reply]
کامندش رو پیدا کردن
بگین برنچ و کامیت مسیجش چی میشه زدم اومده
@AliMD

**Comment by MM25Zamanian** [comment]
اگر بخوایم توی yarn استفاده راه بندازیم این رو برنچ و کامیتش چی میشه ؟؟؟
@AliMD

**Reply by alimd** [reply]
اسم برنچ خیلی مهم نیست چون نمیمونه
کامیت هم میشه مثلا
chore: config git after package install

**Reply by MM25Zamanian** [reply]
میشه روی نکست بزنم؟

**Reply by alimd** [reply]
درضمن با توجه به کثیف شدن راه حل میشه بیخیال بشیم و فقط در contirbute داکیومنت کنیم کامند گیتش رو

**Reply by MM25Zamanian** [reply]
نظری ندارم
هرچی شما بگین
